frontend/src/app/main/ui/workspace/sidebar/options/rows/color_row.scss

```diff
@@ -34,156 +39,289 @@
-    border: deprecated.$s-1 solid var(--input-border-color-hover);
-    .color-bullet-wrapper,
-    .color-name,
-    .detach-btn,
-    .color-input-wrapper {
-      background-color: var(--input-background-color-hover);
-    }
     .detach-btn {
-      display: flex;
+      display: grid;
     }
     &.editing {
-      background-color: var(--input-background-color-active);
-      .color-bullet-wrapper,
-      .color-name,
-      .detach-btn,
-      .color-input-wrapper {
-        background-color: var(--input-background-color-active);
-      }
+      --color-name-wrapper-background-color: var(--color-background-primary);
     }
+
     &:focus,
     &:focus-within {
-      background-color: var(--input-background-color-focus);
-      border: deprecated.$s-1 solid var(--input-border-color-focus);
+      --color-name-wrapper-background-color: var(--color-background-tertiary);
+      --color-name-wrapper-boder-color: var(--color-accent-primary);
     }
   }
 
   &:focus,
   &:focus-within {
-    background-color: var(--input-background-color-focus);
-    border: deprecated.$s-1 solid var(--input-border-color-focus);
+    --color-name-wrapper-background-color: var(--color-background-tertiary);
+    --color-name-wrapper-boder-color: var(--color-accent-primary);
     &:hover {
-      background-color: var(--input-background-color-hover);
-      border: deprecated.$s-1 solid var(--input-border-color-focus);
+      --color-name-wrapper-background-color: var(--color-background-quaternary);
     }
   }
 
   &.editing {
-    background-color: var(--input-background-color-active);
+    --color-name-wrapper-background-color: var(--color-background-primary);
     &:hover {
-      border: deprecated.$s-1 solid var(--input-border-color-active);
+      --color-name-wrapper-boder-color: var(--color-accent-primary);
     }
   }
 }
 
+.detach-btn {
+  display: none;
+  background: var(--color-name-wrapper-background-color);
+}
+
+.color-input-wrapper {
+  @include t.use-typography("body-small");
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  height: $sz-28;
+  width: 100%;
+  padding: 0;
+  margin: 0;
+  background-color: var(--color-name-wrapper-background-color);
+  color: var(--color-name-wrapper-foreground-color);
+  border-radius: 0;
+}
+
+.color-name {
+  @include t.use-typography("body-small");
+  @include textEllipsis;
+  flex-grow: 1;
+  padding-inline: px2rem(6);
+  border-radius: $br-8;
+  color: var(--color-name-wrapper-foreground-color);
+}
+
+.color-bullet-wrapper {
+  display: flex;
+  align-items: center;
+  position: relative;
+  height: $sz-28;
+  padding: 0 var(--sp-xxs) 0 var(--sp-s);
+  border-radius: $br-8 0 0 $br-8;
+  background-color: transparent;
+  &:hover {
+    background-color: transparent;
+  }
+}
+
+.color-input {
+  @include textEllipsis;
+  border: none;
+  background: none;
+  outline: none;
+  height: $sz-28;
+  width: 100%;
+  flex-grow: 1;
+  margin: var(--sp-xxs) 0;
+  padding: 0 0 0 px2rem(6);
```

Comment:
our DS is using x4 sizing properties and expects always variables.
Please ensure that this is design intended flaky padding, otherwise use a x4 padding value